homotopy-graphics/src/style.rs

```diff
@@ -0,0 +1,22 @@
+use homotopy_core::Generator;
+
+pub trait GeneratorStyle {
+    fn label(&self) -> Option<String>;
+    fn shape(&self) -> Option<VertexShape>;
+    // TODO(thud): migrate color-related code to here for consistency
+}
+
+pub trait GeneratorStyles<T: GeneratorStyle> {
```

Comment:
This is a great way to avoid the dependency problem, well done!

Have you considered moving the type `T` inside the trait to avoid having so many generics everywhere, i.e.

```rust
pub trait GeneratorStyles {
    type T: GeneratorStyle;

    fn generator_style(&self, g: Generator) -> Option<&Self::T>;
}
```

Also could we find a different name? How about `Signature` or `SignatureStylesheet`?

Reply:
In [c9f325d](https://github.com/homotopy-io/homotopy-rs/pull/588/commits/c9f325ddc6648f59b6bda531f44e4f60fe206ee2), I have renamed `GeneratorStyles` to `SignatureStyleData`.

I think `SignatureStylesheet` might get confusing quickly in the current codebase and `Signature` is perhaps too vague here.

If we merged `SignatureStylesheet` code into `SignatureStyleData`, then we could think about another rename.